### PR TITLE
MONGOID-5237 Add Feature Flag: _matches? should use millisecond precision when comparing Time objects

### PR DESCRIPTION
**Pull request by Neilshweky**
MONGOID-5237

### CODE REVIEW

**Comment by johnnyshields** [comment]
No, no, no.

**Comment by Neilshweky** [comment]
> No, no, no.

Hey Johnny, the reason this one was put behind a feature flag is because more precision means that fewer things will be matched on a call to `_matches?`. This is a breaking change because a user could be relying on something that matches in 7.3.3 and then in 7.4 it no longer does. This feels like the type of breaking change that we want to make only in a major version.

**Comment by johnnyshields** [comment]
It's great that Mongoid team is now mindful of creating a stable upgrade path, but many of these feature flags seem extreme.

It's super unlikely that any app is going to be affected by this change in the real world.

File: docs/reference/configuration.txt [hunk 1]
```diff
@@ -260,6 +260,11 @@ for details on driver options.
       # error. (default: true)
       belongs_to_required_by_default: true
 
+      # In Mongoid 7.3.3 and earlier, the _matches? function does not use
+      # millisecond precision when comparing time objects. Turning on this
+      # feature flag enables this. (default: false)
```
[the remark this answers is not in the document]
Comment (Neilshweky) [reply docs/reference/configuration.txt]: done

**Review by comandeo — APPROVED** [review]

File: docs/reference/configuration.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Neilshweky) [reply docs/reference/configuration.txt]: done!

**Review by p-mongo — APPROVED** [review]

**Pull request merged**